packages/core/src/httpClient/basicHttpClient.ts

```diff
@@ -197,11 +205,17 @@ export class BasicHttpClient {
     const responseHandler = BasicHttpClient.getResponseHandler<ResponseType>(
       request.responseType
     );
-    const data = await responseHandler(res);
+    // Cloning to fallback on text response if response is failing the responsehandler.
+    // node-fetch < 3.0 will hang on clone() for large responses, that is why the parallel promises https://github.com/node-fetch/node-fetch#custom-highwatermark
+    const resClone = res.clone();
+    const [data, textFallback] = await Promise.all([
```

Comment:
don't the other handlers need to be updated for this as well?

Reply:
to extend on this: what happens if we get a 429 when requesting a file?

eg. using `arrayBufferResponseHandler`.

Reply:
is this change still needed? 
![image](https://user-images.githubusercontent.com/7851860/144218910-4ad3cf63-94ff-4f4b-bec3-7433e008c7c5.png)

Reply:
No, forgot to revert that, now it should be reverted. Thanks!

Reply:
Any final thoughts @andersfylling ?

Reply:
I'm going through the error handling now. Tests (fetch mock) would be great, to actually verify that this fixes the issues.

Reply:
Have added a simple test, that fails without the fix, and works with the fix 👍🏻